File: apps/cargo-tracker/src/contexts/estimation/domain/repository/estimate-repository.ts
import type { Estimate } from '../model/estimate.js';

/** 見積リポジトリ出力ポート */
export interface EstimateRepository {
  save(estimate: Estimate): Promise<number>;
  findByEstimateId(estimateId: string): Promise<Estimate | null>;
  findAll(): Promise<Estimate[]>;
}

File: docs/article/source/scala/apps/cargo-tracker/app/cargotracker/estimation/domain/model/repositories/EstimateRepository.scala
package cargotracker.estimation.domain.model.repositories

import cargotracker.estimation.domain.model.aggregates.Estimate
import cargotracker.estimation.domain.model.valueobjects.EstimateId
trait EstimateRepository:
  def findById(estimateId: EstimateId): Option[Estimate]
  def findAll(): Seq[Estimate]
  def save(estimate: Estimate): Unit
